qiskit_experiments/characterization/t1_analysis.py

```diff
@@ -75,7 +78,7 @@ def _run_analysis(
         offset_bounds=None,
         plot=True,
         ax=None,
-    ) -> Tuple[List[AnalysisResult], List["matplotlib.figure.Figure"]]:
+    ) -> Tuple[List[AnalysisResultData], Optional[List["matplotlib.figure.Figure"]]]:
```

Comment:
Is Optional correct type hint when it returns either List or None. Maybe we should change to
```suggestion
    ) -> Tuple[List[AnalysisResultData], List["matplotlib.figure.Figure"]]:
```
and return `[]` instead of `None` if there is no mpl.